scripts/globals/weaponskills/energy_drain.lua

```diff
@@ -1,11 +1,33 @@
 -----------------------------------
 -- Energy Drain
 -----------------------------------
+require("scripts/globals/status")
+require("scripts/settings/main")
+require("scripts/globals/weaponskills")
+-----------------------------------
 local weaponskill_object = {}
 
 weaponskill_object.onUseWeaponSkill = function(player, target, wsID, tp, primary, action, taChar)
-   -- TODO: Should steal MP based on TP and not wake the mob involved.
-    return 1, 0, false, 65
+   -- TODO: Should not wake the mob involved. Not tested on sleeping mob
+
+    local damage = 60 -- Base mp absorbtion from http://wiki.ffo.jp/html/686.html saying it was between 60 and 120 and always higher than Energy Steal
+
+    -- params.ftp100 = 1 params.ftp200 = 1.5 params.ftp300 = 2 -- to follow earlier statement of been always higher than Energy Steal
+    -- params.mnd_wsc = 1
+
+    local damagemod = damage * ((tp - 1000) / 2000 + 1) -- Simplified formula for getting difference in tp return from 1000 tp to 3000 tp
+
+    if xi.settings.USE_ADOULIN_WEAPON_SKILL_CHANGES then
+        local attMind = player:getStat(xi.mod.MND)
+        damagemod = damagemod + attMind * 1
+    end
+
+    damagemod = damagemod * xi.settings.WEAPON_SKILL_POWER
+    local mpReturn = target:addMP(-damagemod)
+    player:addMP(mpReturn) -- this isnt the right way but the only way I could figure out
+    -- to do make undead undrainable
```

Comment:
Looks like the way to check if something is undead is with `target:isUndead()` , cannibal blade uses it for the same purpose but with drain

Reply:
I will add and test it a bit later thanks

Reply:
Added the check to undead and tested on bogy and that was just what I needed.